opendatahub-io/odh-dashboard
# Update modals and generate meaningful k8s name for images

Closes #1433 #1435 #1388 #1326 

## Description


This PR did the following things:
1. Add the comment back as mentioned in https://github.com/opendatahub-io/odh-dashboard/pull/1506#discussion_r1265789708 since we are not going to fix it in this phase and I removed it by mistake
2. Enhance the status toggle of `Enable` column in the table, for the error images, it will set default to disabled and the user cannot enable it. Also, add the sort to the `Enable` column as @vconzola requested here https://github.com/opendatahub-io/odh-dashboard/pull/1506#issuecomment-1633095279

<img width="698" alt="Screenshot 2023-07-17 at 4 38 58 PM" src="https://github.com/opendatahub-io/odh-dashboard/assets/37624318/36117267-7fd2-4f3d-a12c-e7ca666c3f12">

3. Generate a meaning k8s name for the image stream that user imports, before the change, it was `byon-${timestamp}`, but now it's `custom-${translated-k8s-name}` where `translated-k8s-name` is the translation of the display name that user inputs. Also, add a tooltip next to the image name so the user could easily locate the image stream resource on the cluster.

<img width="698" alt="Screenshot 2023-07-17 at 4 45 23 PM" src="https://github.com/opendatahub-io/odh-dashboard/assets/37624318/74381bfb-f995-407c-9877-0c6b1c1854d4">

4. Refactor the Import/Update modal of the custom notebook images, change the labels, and add helper text/tooltips based on the mockups

<img width="940" alt="Screenshot 2023-07-17 at 4 47 59 PM" src="https://github.com/opendatahub-io/odh-dashboard/assets/37624318/698e89b1-5bb9-45ae-9684-9dc54d073242">

**Highlights**: Refactor the table for adding/editing packages and software in the modal. Take the software as an example, when the user starts to add software or edit software, he will be in the edit mode, which only allows the user to confirm or abandon the current change (the user cannot edit multiple lines at the same time and cannot submit or exit the modal). 

What's more, when the user hits `Enter` in the edit mode, it will automatically confirm the current row and move to the next row. If the row is already the last row, it will create a new entry, which helps the user to quickly input and move on without clicking with the mouse every time. When the user hits `Esc` in the edit mode, it will abandon the current change of the row and exit the edit mode.

Edit mode: close button hidden, Submit and Cancel button disabled, other rows cannot be edited and removed, either
<img width="940" alt="Screenshot 2023-07-17 at 5 02 44 PM" src="https://github.com/opendatahub-io/odh-dashboard/assets/37624318/408b0dd1-f5b5-462e-95b4-66daca4cf065">
Non-edit mode: You can do anything you want
<img width="940" alt="Screenshot 2023-07-17 at 5 03 10 PM" src="https://github.com/opendatahub-io/odh-dashboard/assets/37624318/d54600b3-cc06-4f7b-a50a-289b723449bc">


## How Has This Been Tested?



1. Try to create some images with the correct path (like `quay.io/opendatahub/workbench-images:jupyter-minimal-ubi8-python-3.8-pr-89`) and incorrect path (like `quay.io/opendatahub/workbench-images:invalid-tag`) and you could check the status of the toggle on the `Enable` column. Try to create more and sort by the `Enable` column
2. Try to import/edit the image in every situation, play around with the edit mode of the software/packages

## Test Impact


TBD

## Request review criteria:



Self checklist (all need to be checked):

If you have UI changes: 


After the PR is posted & before it merges:
